src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -59,6 +59,41 @@ object SnowflakeTelemetry {
     }
   }
 
+  private var cachedSparkLibraries: Seq[String] = Seq.empty
+
+  private[snowflake] def getSparkLibraries: Seq[String] = {
+    if (cachedSparkLibraries.isEmpty) {
+      try {
+        val classNames = Thread.currentThread()
+          .getStackTrace
+          .reverse // reverse to make caller on the head
+          .map(_.getClassName)
+          .map(_.replaceAll("\\$", "")) // Replace $ for Scala object class
+
+        // Skip known libraries and user's application package
+        val knownLibraries = Seq("java.", "scala.", "net.snowflake.spark.snowflake.") ++
+          Seq(classNames.head.split("\\.").dropRight(1).mkString("", ".", "."))
```

Comment:
can we directly get package name from class?
for example
`this.getClass.getPackage.getName`

Reply:
This doesn't work. I just tried it. it will return `java.lang`.

Even it works, I still prefer to use `_.getClassName` because I want to filter them with `java.` /`scala.` instead of `java`/`scala`.